src/components/shared/RichTextEditor/RichTextEditor.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { useEffect } from 'react';
+import { useFormContext } from 'react-hook-form';
+import { EditorContent, Editor } from '@tiptap/react';
+import { Editor as CoreEditor } from '@tiptap/core';
+import classnames from 'classnames';
+
+import { DEFAULT_CHARACTER_LIMIT } from '~hooks/useRichTextEditor';
+
+import Toolbar from './Toolbar';
+
+import styles from './RichTextEditor.css';
+
+const displayName = 'RichTextEditor';
+
+export interface RichTextEditorProps {
+  className?: string;
+  content?: string;
+  disabled?: boolean;
+  editor: Editor;
+  limit?: number;
+  name: string;
+}
+
+const RichTextEditor = ({
+  className,
+  content = '',
+  editor,
+  disabled = false,
+  limit = DEFAULT_CHARACTER_LIMIT,
+  name,
+}: RichTextEditorProps) => {
+  const {
+    getFieldState,
+    setValue,
+    formState: { isSubmitting },
+  } = useFormContext();
+  const { error: contentError, isTouched: contentTouched } =
+    getFieldState(name);
+
+  useEffect(() => {
```

Comment:
Tiptap does not expose the editors state directly. Instead, you need to hook into events that the base class emits. 

We hook into the "selectionUpdate" event which fires whenever the editor content changes, and also the "blur" event.

This needs to be done inside an effect to ensure clean up happens on dismount and also to avoid setting the form state (via setValue) while the `RichTextEditor` is being rendered.